### PR TITLE
Change FormFactorDot. Suppress warnings. Edit FFCatalog.

### PR DESCRIPTION
**Pull request by jwuttke**
- Suppress warnings from swig-generated code
- Eliminate several other warnings
- FFCatalog: Provide short intro to ripples; start chapter on soft particles
- FormFactorDot: Add 'radius' argument to get dimensions and intensity scaling right

### CODE REVIEW

File: Tests/Functional/GUI/GUIStandardTest/GUIStandardTest.cpp [hunk 1]
```diff
@@ -49,14 +49,11 @@ std::unique_ptr<Simulation> createDomainSimulation(const Simulation& origin)
 bool GUIStandardTest::runTest()
 {
     m_reference_simulation->runSimulation();
-    auto ref_result = m_reference_simulation->result();
+    const SimulationResult& ref_result = m_reference_simulation->result();
```
Comment (gpospelov) [inline Tests/Functional/GUI/GUIStandardTest/GUIStandardTest.cpp:52]: I find the syntax with `const` reference to a temporary object more confusing. It is allowed, but confusing, to my taste. My suggestion would be to stick to [almost always auto](https://blog.galowicz.de/2016/03/23/const_reference_to_temporary_object/) in newly written code, while leaving old code intac.

File: Core/Tools/PythonCore.h [hunk 2]
```diff
@@ -28,7 +28,11 @@
 #define NO_IMPORT_ARRAY
 #include <numpy/arrayobject.h>
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```
Comment (gpospelov) [inline Core/Tools/PythonCore.h]: That's nice that old Macros.h is removed, it was generating warnings in my case too. For the future, however, we could consider another, more modern version of similar macros, to centralize Warning suppression in the compiler independent way. There is a nice [post](https://www.fluentcpp.com/2019/08/30/how-to-disable-a-warning-in-cpp/) about the problem.

File: Core/CMakeLists.txt [hunk 3]
```diff
@@ -66,6 +66,14 @@ if(BORNAGAIN_PYTHON)
             list(APPEND SWIG_FLAGS "-I${dir}")
         endforeach(dir)
 
+        if ((CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+                OR (CMAKE_CXX_COMPILER_ID STREQUAL "GNU"))
+            # suppress warnings from auto-generated code (last updated for Swig 4.0.1)
+            set_source_files_properties(${AUTO_DIR}/libBornAgainCore_wrap.cpp
+                PROPERTIES COMPILE_OPTIONS
+                "-Wno-unused-parameter;-Wno-missing-field-initializers;-Wno-sometimes-uninitialized;-Wno-deprecated-declarations")
+        endif()
```
Comment (gpospelov) [inline Core/CMakeLists.txt:75]: This is nice!

File: GUI/coregui/Views/SampleDesigner/DesignerHelper.cpp [hunk 4]
```diff
@@ -37,7 +37,7 @@ QGradient DesignerHelper::getLayerGradient(const QColor& color, const QRectF& re
     result.setColorAt(0.5, c.light(200));
     result.setColorAt(1, c.dark(150));
 #endif
-    return result;
+    return std::move(result);
```
Comment (gpospelov) [inline GUI/coregui/Views/SampleDesigner/DesignerHelper.cpp:40]: Changes do not match the commit message. Also move is not necessary in this context, to my opinion.

File: Tests/Functional/GUI/GUIStandardTest/GUIStandardTest.cpp (hunk 1 above)
Comment (gpospelov) [reply Tests/Functional/GUI/GUIStandardTest/GUIStandardTest.cpp:52]: Gave a wrong reference to [almost always auto](https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/)

File: GUI/coregui/Views/SampleDesigner/DesignerHelper.cpp (hunk 4 above)
Comment (jwuttke) [reply GUI/coregui/Views/SampleDesigner/DesignerHelper.cpp:40]: Was prompted by clang warning.

Comment (gpospelov) [reply GUI/coregui/Views/SampleDesigner/DesignerHelper.cpp:40]: Good to know, will investigate, but probably, clang is wrong :). Thanks.

**Pull request merged**